src/resources/parser/texture/ktx.js

```diff
@@ -73,99 +92,86 @@ class KtxParser {
+        // check magic bits
+        if (IDENTIFIER[0] !== dataU32[0] ||
+            IDENTIFIER[1] !== dataU32[1] ||
+            IDENTIFIER[2] !== dataU32[2]) {
             // #if _DEBUG
             console.warn("Invalid definition header found in KTX file. Expected 0xAB4B5458, 0x203131BB, 0x0D0A1A0A");
             // #endif
             return null;
         }
 
+        // unpack header info
         const header = {
-            endianness: headerU32[3], // todo: Use this information
-            glType: headerU32[4],
-            glTypeSize: headerU32[5],
-            glFormat: headerU32[6],
-            glInternalFormat: headerU32[7],
-            glBaseInternalFormat: headerU32[8],
-            pixelWidth: headerU32[9],
-            pixelHeight: headerU32[10],
-            pixelDepth: headerU32[11],
-            numberOfArrayElements: headerU32[12],
-            numberOfFaces: headerU32[13],
-            numberOfMipmapLevels: headerU32[14],
-            bytesOfKeyValueData: headerU32[15]
+            endianness: dataU32[3], // todo: Use this information
+            glType: dataU32[4],
+            glTypeSize: dataU32[5],
+            glFormat: dataU32[6],
+            glInternalFormat: dataU32[7],
+            glBaseInternalFormat: dataU32[8],
+            pixelWidth: dataU32[9],
+            pixelHeight: dataU32[10],
+            pixelDepth: dataU32[11],
+            numberOfArrayElements: dataU32[12],
+            numberOfFaces: dataU32[13],
+            numberOfMipmapLevels: dataU32[14],
+            bytesOfKeyValueData: dataU32[15]
         };
 
+        // don't support volume textures
         if (header.pixelDepth > 1) {
             // #if _DEBUG
             console.warn("More than 1 pixel depth not supported!");
             // #endif
             return null;
         }
 
-        if (header.numberOfArrayElements > 1) {
+        // don't support texture arrays
+        if (header.numberOfArrayElements !== 0) {
             // #if _DEBUG
             console.warn("Array texture not supported!");
             // #endif
             return null;
         }
 
-        if (header.glFormat !== 0) {
-            // #if _DEBUG
-            console.warn("We only support compressed formats!");
-            // #endif
-            return null;
-        }
-
+        // only support subset of pixel formats
         if (!KNOWN_FORMATS[header.glInternalFormat]) {
             // #if _DEBUG
             console.warn("Unknown glInternalFormat: " + header.glInternalFormat);
             // #endif
             return null;
         }
 
-        // Byte offset locating the first byte of texture level data
-        let offset = (16 * 4) + header.bytesOfKeyValueData;
+        // offset locating the first byte of texture level data
+        let offset = 16 + header.bytesOfKeyValueData / 4;
 
+        const format = KNOWN_FORMATS[header.glInternalFormat];
```

Comment:
move this few lines up to avoid doing it twice

Reply:
yes! thanks